### DISCUSSION TITLE
General Standards

### DISCUSSION THREAD

**Discussion by mpadge** [ropensci/statistical-software-review-book#44] [Standards]
## General Standards for Statistical Software

This is a place to discuss our [*General Standards for Statistical Software*](https://stats-devguide.ropensci.org/standards.html#general-standards). Our standards are community-led, and are intended to evolve in response to ongoing community engagement and feedback. Any suggestions for improvements, modifications, or edits small and large are welcome :clap: 

Participation in these discussions are guided by our [Code of Conduct](https://ropensci.org/code-of-conduct/).

**Comment by cforgaci** [comment]
While checking [our package](https://github.com/CityRiverSpaces/rcrisp) against standard G1.4a, I was wondering about:

1. the use of the `@noRd` in *all* non-exported functions, and
2. the usefulness of documenting functions tagged with `@noRd`.

Our approach so far has been to use `@keywords internal` whenever we wanted to generate documentation for non-exported functions and a simple `@noRd` whenever documentation was not needed.

Regarding the first point, if we add `@noRd` to functions tagged with `@keywords internal` (which is equivalent with replacing `@keywords internal` with `@noRd`), the internal documentation will not be visible anymore and that might be undesirable. Is this necessary?

As for the second point, what is expected with "documented in standard `roxygen2` format"? Is simply the use of `@noRd` tag considered to be sufficient?

Any clarification would be highly appreciated! Thanks in advance!

**Reply by mpadge** [reply]
Good point @cforgaci! The `@noRd` recommendation just follows the advice in [our general _Dev Guide_](https://devguide.ropensci.org/pkg_building.html#roxygen-2-use), which only mentions `@noRd`. Of course `@keywords internal` should also be encouraged, and is often better for reasons you describe. Please feel very welcome to open an issue, or submit a direct PR, to https://github.com/ropensci/dev_guide/. The line you'll need to update is [here](https://github.com/ropensci/dev_guide/blob/3d29603f8e0236b6602bd32f6fb57185d92c5fdc/pkg_building.Rmd#L255). We should then also update [Standard G1.4](https://github.com/ropensci/dev_guide/blob/3d29603f8e0236b6602bd32f6fb57185d92c5fdc/pkg_building.Rmd#L255) (via [source here](https://github.com/ropensci/statistical-software-review-book/blob/14030432180e50f93200e0d256c3476cbe6d6e05/standards/general.Rmd#L128-L133)). Your contributions to both would be very welcome! Thanks for your engagement here. And ping @maelle
